pytket_qir/converter.py

```diff
@@ -0,0 +1,357 @@
+                self._successors[block_name] = []
+                self.edges[block_name] = {}
+        # Sort the DAG.
+        sorted_dag = topological_sort(self._successors)
+        # Update (sort) the list of successors.
+        for block, succs in sorted_dag.items():
+            if len(succs) > 1:
+                f = (
+                    lambda el: el == self._successors[block][0]
+                    or el == self._successors[block][1]
+                )
+                self._successors[block] = list(filter(f, self._successors))
+
+    @property
+    def predecessors(self):
+        return self._predecessors
+
+    @predecessors.setter
+    def predecessors(self, value):
+        self._predecessors = OrderedDict()
+        reversed_succs = reversed(self.successors)
+        reversed_succs_list = list(reversed_succs)
+        for curr_block_index, curr_block_name in enumerate(reversed_succs_list):
+            predecessors = []
+            for next_block_name in reversed_succs_list[curr_block_index + 1 :]:
+                if curr_block_name in self.successors[next_block_name]:
+                    predecessors.append(next_block_name)
+            self._predecessors[curr_block_name] = predecessors
+
+    @property
+    def conditions(self):
+        return self._conditions
+
+    @conditions.setter
+    def conditions(self, value):
+        self._conditions = value
+
+    @property
+    def edges(self):
+        return self._edges
+
+    @edges.setter
+    def edges(self, value):
+        self._edges = value
+
+    @property
+    def local_conditions(self):
+        return self._local_conditions
+
+    @local_conditions.setter
+    def local_conditions(self, value):
+        self._local_conditions = value
+
+    @property
+    def cfg(self):
+        return self._cfg
+
+    @cfg.setter
+    def cfg(self, value) -> None:
+        # import pdb; pdb.set_trace()
+        self.conditions = {}
+        self.edges = {}
+        self.local_conditions = {}
+        self.successors = {}
+        self.predecessors = {}
+
+        self._cfg = value
+        if not value:  # Checking for empty input dict.
+            for block_name, succs in self.successors.items():
+                block_inst = Block(
+                    name=block_name,
+                    succs=succs,
+                    preds=self.predecessors[block_name],
+                    composition=[block_name],
+                    visited=False,
+                )
+                # f = lambda b: b if b.condition else None
+                self._cfg[block_name] = block_inst
+
+    def cfg_to_circuit(self) -> Circuit:
+        main_circuit = Circuit(self.parser.qubits, self.parser.bits + 2)
+        # Setting condition and local condition to True
+        # for the predecessor of the entry block.
+        main_circuit.add_c_setbits([1, 1], [self.parser.bits, self.parser.bits + 1])
+        condition_bit = main_circuit.get_c_register("c")[self.parser.bits]
```

Comment:
Are there any plans how to invert this bits when converting back to qir? Have you thought about putting special label on them?

Reply:
It could be helpful to mark this somehow to make it easier to detect them later?

Reply:
Hm, not sure I follow you. To mark what exactly ?

Reply:
To mark that this bit as been created to store the block condition?